documentation/modules/deploying/proc-deploy-migrate-kraft.adoc

```diff
@@ -0,0 +1,257 @@
+but controllers are managing metadata in Kafka and ZooKeeper. 
+Brokers are rolled a second time to remove the flag to enable migration.
+* `KRaftPostMigration` is the state when KRaft mode is enabled for brokers and there is no ZooKeeper involvement. 
+Controllers are still managing in Kafka and ZooKeeper.
+You can xref:proc-deploy-migrate-kraft-rollback-{context}[roll back from this point].
+--
++
+The migration status is also represented in the `status.kafkaMetadataState` property of the `Kafka` resource. 
+
+. Enable KRaft in the `Kafka` resource configuration by setting the `strimzi.io/kraft` annotation to `enabled`:
++
+[source,shell]
+----
+kubectl annotate kafka my-cluster strimzi.io/kraft: enabled
+----
++
+.Enabling KRaft migration
+[source,yaml,subs="+attributes"]
+----
+apiVersion: {KafkaApiVersion}
+kind: Kafka
+metadata:
+  name: my-cluster
+  namespace: my-project
+  annotations:
+    strimzi.io/kraft: enabled 
+# ...
+----
++
+WARNING: Rollback cannot be performed after enabling KRaft.
+
+. Check the status of the move to full KRaft mode:
++
+[source,shell]
+----
+kubectl get kafka my-cluster -n my-project -w
+----
++
+.Updates to the metadata state
+[source,shell]
+----
+NAME        ...  METADATA STATE
+my-cluster  ...  Zookeeper
+my-cluster  ...  KRaftMigration
+my-cluster  ...  KRaftDualWriting
+my-cluster  ...  KRaftPostMigration
+my-cluster  ...  PreKRaft
+my-cluster  ...  KRaft             
+----
++
+--
+* `PreKRaft` is the state when all ZooKeeper-related configurations and resources have been automatically deleted.
+* `KRaft` is the final state (after the controllers have rolled) when the KRaft migration is finalized.
+--
+
+[id='proc-deploy-migrate-kraft-rollback-{context}']
+.Performing a rollback on the migration
+
+Before the migration is finalized by enabling KRaft in the `Kafka` resource,  and the state has moved to the `KRaft` state, you can perform a rollback operation as follows:
+
+. Apply the `strimzi.io/kraft: rollback` annotation to the `Kafka` resource to roll back the brokers.
++
+[source,shell]
+----
+kubectl annotate kafka my-cluster strimzi.io/kraft: rollback
+----
++
+.Rolling back KRaft migration
+[source,yaml,subs="+attributes"]
+----
+apiVersion: {KafkaApiVersion}
+kind: Kafka
+metadata:
+  name: my-cluster
+  namespace: my-project
+  annotations:
+    strimzi.io/kraft: rollback 
+# ...
+----
++
+The migration process must be in the `KRaftPostMigration` state to do this. 
+The brokers are rolled back so that they can be connected to ZooKeeper again and the state returns to `KRaftDualWriting`.
+
+. Delete the controllers node pool:
++
+[source,shell]
+----
+kubectl delete KafkaNodePool controller -n my-project
+----
+
+. Apply the `strimzi.io/kraft: disabled` annotation to the `Kafka` resource to return the metadata state to `ZooKeeper`.
++
+[source,shell]
+----
+kubectl annotate kafka my-cluster strimzi.io/kraft: disabled
```

Comment:
```suggestion
kubectl annotate kafka my-cluster strimzi.io/kraft="disabled" --overwrite
```

Reply:
Yeah good catch @fvaleri , in all commands we would need the `--overwrite` because the annotation could be already present on the `Kafka` CR as `disabled` (without overwrite it's not going to work).